apply_kernel.json

```diff
@@ -38,6 +38,26 @@
             },
             "default": 1,
             "optional": true
+        },
+        {
+            "name": "border",
+            "description": "Determines how the data is extended when the filter overlaps the borders. Defaults to fill the border with zeroes.\n\nThe following options are available:\n\n* *numeric value* - fill with a user-defined constant number `n`: `nnnnnn|abcdefgh|nnnnnn` (default, with `n` = 0)\n* `replicate` - repeat the value from the pixel at the border: `aaaaaa|abcdefgh|hhhhhh`\n* `reflect` - mirror/reflect from the border: `fedcba|abcdefgh|hgfedc`\n* `reflect_pixel` - mirror/reflect from the center of the pixel at the border: `gfedcb|abcdefgh|gfedcb`\n* `wrap` - repeat/wrap the image: `cdefgh|abcdefgh|abcdef`",
```

Comment:
> when the filter overlaps the borders

instead of "filter", use "kernel" because that is used in other places?

Maybe also add an option `auto` (which could even be the default): let back-end pick the approach that is most efficient  (e.g. zero-filling for direct calculation, wrap for FFT based calculations)

Reply:
> instead of "filter", use "kernel" because that is used in other places?

Yes, changed!

> Maybe also add an option `auto` (which could even be the default): let back-end pick the approach that is most efficient (e.g. zero-filling for direct calculation, wrap for FFT based calculations)

If really necessary, back-ends can still change the default value in the process description although that decreases interoperability. So `auto` is not really needed, but both options are not recommended anyway.